tutorial/unity_local_realm.md

```diff
@@ -0,0 +1,608 @@
+
+Here you need to add the following line to the `dependencies`:
+
+```json
+"io.realm.unity": "<version-number>"
+```
+
+Replace `<version-number>` with the most recent Realm version found in https://github.com/realm/realm-dotnet/releases and you're all set.
+
+The final `manifest.json` should look something like this:
+
+```json
+{
+ "dependencies": {
+   ...
+   "io.realm.unity": "10.3.0"
+ },
+ "scopedRegistries": [
+   {
+     "name": "NPM",
+     "url": "https://registry.npmjs.org/",
+     "scopes": [
+       "io.realm.unity"
+     ]
+   }
+ ]
+}
+```
+
+When you switch back to Unity it will reload the dependencies. If you then open the `Package Manager` again, you should see `Realm` as a new entry in the list on the left:
+
+![Realm in Project Manager](images/05_realm_in_project_manager.jpg)
+
+We can now start using Realm in our Unity project.
+
+## Top-down or bottom-up?
+
+Before we actually start adding Realm to our code we need to think about how we want to achieve this and how the UI and database will interact with each other.
+
+There are basically two options we can choose from: top-down or bottom-up.
+
+The top-down approach would be to have the UI drive the changes. The `Piece` would know about it's database object and whenever a `Piece` is moved it would also update the database with it's new position.
+
+The preferred approach would be bottom-up though. Changes will be applied to the Realm and it will then take care of whatever implications this has on the UI by sending notifications.
+
+Let's first look into the initial setup of the board.
+
+## Setting up the board
+
+The first thing we want to do is to define a Realm representation of our Piece since we cannot save the `MonoBehaviour` directly in Realm. Classes that are supposed to be saved in Realm need to subclass `RealmObject`. The class `PieceEntity` will represent such an object. Note that we can not just duplicate the types from `Piece` since [not all of them can be saved in Realm](https://docs.mongodb.com/realm/sdk/dotnet/data-types/field-types/), like `Vector3` and `enum`.
+
+Add the following scripts to the project:
+
+```c#
+using Realms;
+using UnityEngine;
+
+public class PieceEntity : RealmObject
+{
+    // 1
+    public PieceType PieceType
+    {
+        get => (PieceType)Type;
+        private set => Type = (int)value;
+    }
+
+    // 2
+    public Vector3 Position
+    {
+        get => PositionEntity.ToVector3();
+        set => PositionEntity = new Vector3Entity(value);
+    }
+
+    // 3
+    private int Type { get; set; }
+    private Vector3Entity PositionEntity { get; set; }
+
+    // 4
+    public PieceEntity(PieceType type, Vector3 position)
+    {
+        PieceType = type;
+        Position = position;
+    }
+
+    // 5
+    protected override void OnPropertyChanged(string propertyName)
+    {
+        if (propertyName == nameof(PositionEntity))
```

Comment:
I understand you probably won't need it, but would it make sense to the same for `Type` and `PieceType`, for consistency?

Reply:
We don't need it here that's why we didn't add it. I prefer to not add code that is never run but maybe @nirinchev can add a comment about consistency vs. code coverage.